File: threads/src/test/java/com/sequenia/threads/model/ConsultInfoTest.java
package com.sequenia.threads.model;

import org.junit.Test;

import static org.junit.Assert.*;

/**
 * Created by yuri on 09.09.2016.
 */
public class ConsultInfoTest {

    @Test
    public void testToJson() throws Exception {
        ConsultInfo consultInfo = new ConsultInfo("Test Operator #1","2","все плохо","http:\\/\\/pushservertest.mfms.ru\\/push-test\\/file\\/download\\/jm7upm65581xkd10l2froa60tqg1z035w8rk");
        assertEquals("{\"photoUrl\":\"http://pushservertest.mfms.ru/push-test/file/download/jm7upm65581xkd10l2froa60tqg1z035w8rk\"," +
                "\"name\":\"Test Operator #1\",\"id\":\"2\"," +
                "\"status\":\"все плохо\"}",consultInfo.toJson().toString().replaceAll("\\\\",""));

    }
}

File: threads/src/test/java/im/threads/business/models/ConsultInfoTest.kt
package im.threads.business.models

import com.google.gson.JsonObject
import org.junit.Assert.assertEquals
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner

@RunWith(RobolectricTestRunner::class)
class ConsultInfoTest {

    private val consultInfo = ConsultInfo(
        name = "Test Name",
        id = "Test ID",
        status = "Test Status",
        organizationUnit = "Test Organization Unit",
        photoUrl = "Test Photo URL",
        role = "Test Role"
    )

    @Test
    fun whenToString_thenReturnsCorrectString() {
        val expectedString = """ConsultInfo{
            name='Test Name',
            id=Test ID,
            status='Test Status',
            organizationUnit='Test Organization Unit',
            photoUrl='Test Photo URL',
            role='Test Role'
            }
        """.trim()
        assertEquals(expectedString, consultInfo.toString())
    }

    @Test
    fun whenToJson_thenReturnsCorrectJsonObject() {
        val expectedJsonObject = JsonObject().apply {
            addProperty("name", "Test Name")
            addProperty("status", "Test Status")
            addProperty("id", "Test ID")
            addProperty("photoUrl", "Test Photo URL")
            addProperty("role", "Test Role")
        }
        assertEquals(expectedJsonObject, consultInfo.toJson())
    }
}
